### ISSUE TITLE
bug: table with postgres-cdc connector does not get updated.

### ISSUE THREAD

**Issue by Povilaszva** [risingwavelabs/risingwave#10737]
### Describe the bug

```text
dev=> CREATE TABLE table_test (
    handle character varying,
    key_type text,
    value_type text,
    inserted_at timestamp without time zone,
    PRIMARY KEY (handle)
) WITH (
    connector = 'postgres-cdc',
    hostname = '*.*.*.*',
    port = '5432',
    username = '*******',
    password = '****',
    database.name = 'postgres',
    schema.name = 'public',
    table.name = 'table_test'
);
CREATE_TABLE
dev=> select count(*) from table_test
dev-> ;



  count  
---------
 1293184
(1 row)

```

When new records appear in upstream database this table is not updated.  upstream database is configured following docs:
https://www.risingwave.dev/docs/current/ingest-from-postgres-cdc/

no errors in logs, am i missing something?.

### Error message/log

```text
https://gist.github.com/Povilaszva/d94fd75d7928a9d22af822a60c2b0aa6
```

### The version of RisingWave

v0.19.2

_No response_

**Comment by Povilaszva** [comment]
```text
postgres=# select * from pg_publication;
        pubname         | pubowner | puballtables | pubinsert | pubupdate | pubdelete 
------------------------+----------+--------------+-----------+-----------+-----------
 dbz_publication        |    16385 | f            | t         | t         | t
 mz_source              |       10 | f            | t         | t         | t
 mz_source_transactions |       10 | f            | t         | t         | t
(3 rows)

postgres=# 
postgres=# select * from pg_publication_tables;
        pubname         | schemaname |    tablename    
------------------------+------------+-----------------
 dbz_publication        | public     | transactions
 mz_source_transactions | public     | transactions
 mz_source              | public     | table_metadatas
(3 rows)
```
ran it on upstream postgres.  mz_* is created from other streaming db I've tested.

**Comment by StrikeW** [comment]
Thank you @Povilaszva. From the output, `dbz_publication` only subscribes to the `transactions` so you cannot receive events for the `table_test` table. That's should be a bug in our implementation.
To work around the issue right now, you can choose one option:
1) `drop publication dbz_publication` and `create publication dbz_publication for table table_test` 
or
2) `drop publication dbz_publication` and `create publication dbz_publication for all tables` (require superuser role)

**Comment by rotten** [comment]
It looks like we are not able to select the pubname to use for a connection?

What I think is happening is that the first table we configure creates a new publication in our database automatically called `dbz_publication` and adds the relevant table to it.  After that, any tables we configure, since there already is a publication with that name, it doesn't get updated to include that table.  We get the initial table snapshot, but no subsequent data changes.

So the answer is to go back and manually add the tables to the dbz_publication, or to create a dbz_publication first?

This is confusing behavior even if it isn't a bug.

1.  Ideally we should be able to select a publication name to use for a particular table.
    - In pg 15+ we can specify columns that are published rather than the whole table.
    - We may share a publication with multiple downstream sources,
    - We may want partial change captures (ie, no replication of deletes) for a particular table, whereas other tables we may want all the changes.
2.  If we are required to use that one publication name, then the documentation should be clear that either:
    - You have to set it up in advance.
    - You have to go back and manually add any additional tables after the first one is set up.
3.  Maybe we could have a "create publication on remote database if it doesn't exist" flag when we create the table.

**Comment by rotten** [comment]
I would also add a reminder that you should explicitly add the table to the publication **before** you create the table otherwise you risk missing changes between the initial snapshot and when you enable change capture.

**Comment by StrikeW** [comment]
Hi @rotten, thanks for your sincerely suggestion. I will initiate an enhancement for the part of publication and it would be shipped into our v1.0 release recently.

> So the answer is to go back and manually add the tables to the dbz_publication, or to create a dbz_publication first?

This is just a temporary work around for the problem to unblock your evaluation progress, we will fix it soon.
